Tree.js

```diff
@@ -396,6 +396,19 @@ define([
 			});
 		},
 
+		_onNotification: function (rows, event) {
+			if (event.type === 'delete') {
+				var key;
+				for (key in this._expanded) {
+					var row = this.row(key);
+					if (!row || (!row.data && !row.element)) {
```

Comment:
They are not. I was being thick and missed the fact that the deleted object's ID is in the delete event.